sql-plugin/src/main/scala/org/apache/spark/sql/rapids/AggregateFunctions.scala

```diff
@@ -724,11 +782,19 @@ case class GpuCollectList(
  * with the CPU version and automated checks.
  */
 case class GpuCollectSet(
-    childExpression: Expression,
+    child: Expression,
     mutableAggBufferOffset: Int = 0,
     inputAggBufferOffset: Int = 0)
     extends GpuCollectBase[CollectSetAggregation] {
 
+  override lazy val updateExpressions: Seq[GpuExpression] = new CudfCollectSet(inputBuf) :: Nil
+
+  override lazy val mergeExpressions: Seq[GpuExpression] = new CudfMergeSets(outputBuf) :: Nil
```

Comment:
I'm thinking of totally removing `CudfCollectSet` and `CudfMergeSet`. We just call `CudfCollectList` in `updateExpressions`, and `CudfMergeLists` in `mergeExpressions`. During `evaluateExpression`, we call `cudfDropListDuplicates` on the merged lists.